src/pages/ReportDetailsPage.js

```diff
@@ -93,15 +100,29 @@ function ReportDetailsPage(props) {
             return items;
         }
 
-        if (participants.length) {
+        if ((!isUserCreatedPolicyRoom && participants.length) || (isUserCreatedPolicyRoom && isPolicyMember)) {
             items.push({
                 key: CONST.REPORT_DETAILS_MENU_ITEM.MEMBERS,
                 translationKey: 'common.members',
                 icon: Expensicons.Users,
                 subtitle: participants.length,
                 isAnonymousAction: false,
                 action: () => {
-                    Navigation.navigate(ROUTES.REPORT_PARTICIPANTS.getRoute(props.report.reportID));
+                    if (isUserCreatedPolicyRoom && Permissions.canUsePolicyRooms(props.betas)) {
```

Comment:
```suggestion
                    if (isUserCreatedPolicyRoom) {
```

The policy check should be done on the page level and not on the buttons. Adjust the `FullPageNotFoundView` `shouldShow` prop.

Reply:
We don't want to display the full page not found view in these situations. We just want to decide where the button takes us. Is there a problem?

Reply:
It does not make much sense that a report is policy room and you are being redirected to the wrong page (report participants)

Why the not found view is not a good fit here? What is the expected behavior if you try to access a policy room and you don't have usePolicyRooms permission?

Reply:
Cleared up in DMs